projects/element-ng/chat-messages/si-attachment-list.component.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) Siemens 2016 - 2025
+ * SPDX-License-Identifier: MIT
+ */
+import { booleanAttribute, Component, inject, input, output, TemplateRef } from '@angular/core';
+import { SiIconComponent } from '@siemens/element-ng/icon';
+import { SiModalService } from '@siemens/element-ng/modal';
+import { SiTranslatePipe, t } from '@siemens/element-translate-ng/translate';
+
+/**
+ * Attachment item interface for file attachments in chat messages.
+ *
+ * @experimental
+ */
+export interface Attachment {
+  /** File name */
+  name: string;
+  /** Optionally show a preview of the attachment by providing a template that is shown in a modal when clicked (optional) */
+  previewTemplate?: TemplateRef<any> | (() => TemplateRef<any>);
+  /** Optional unique identifier for the attachment */
+  id?: string;
```

Comment:
I think we do not need the id. We are giving back the object on events. Please remove.

Reply:
Still do not agree that checking by identity is a good solution for this, but removed.